mood-analysis.rb

```diff
@@ -28,5 +29,60 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+def strip_punctuation(words)
+  # 2. To make the results a little more accurate, let's write and utilize a method called `strip_punctuation` to 
+  # strip out the punctuation that affects the results. Namely, remove  exclamation marks (!), periods (.), commas (,), 
+  # and hashtags (#).
+  words.gsub(/[^\/A-Za-z0-9\s]/i, "")
+end
+
+# "smiley collector": iterate a corpus, analyze its mood, store return in an array.
+def print_all_moods(words)
+  words.collect { |statement| statement[0..4] + " " + analyze_mood(statement)}
+end
+
+def happy_days(words)
+  # Write a method called `happy_days` to determine how many logged entries it takes until there have 
+  # been three :-) happy days.
+
+  total_days = 0
+  hap_days = 0
+
+  words.each do |statement|
+    total_days += 1
+    if analyze_mood(strip_punctuation(statement)) == ":-)"
+      hap_days += 1
+    end
+    break if hap_days == 3
+  end
+
+  return "It takes #{total_days} days for 3 happy days to occur." if hap_days == 3
+  return "There were not 3 happy days."
+
+end
+
+def overall_mood(words)
+  feels = {
+          happy: [],
+          sad: [],
+          meh: []
+  }
+
+  words.each do |statement|
+    if analyze_mood(statement) == ":-)"
+      feels[:happy] << ":-)"
+    elsif analyze_mood(statement) == ":-("
+      feels[:sad] << ":-("
+    else
+      feels[:meh] << ":-|"
+    end
+  end
```

Comment:
This works, but isn't super expandable without changing the code. Using a hash to count an undetermined number of things is actually fairly common in whiteboarding problems, so it's an important concept to understand whether or not it's the solution that you came up with on you own.

Here's my overall_mood method for reference:
```
def overall_mood(entries)
  return nil if entries.length == 0
  emoticons = Hash.new(0)
  entries.each do |entry|
    emoticon = analyze_mood(entry)
    emoticons[emoticon] += 1
  end
  return emoticons.max_by{|k,v| v}[0]
end
```